### PR TITLE
feat: registry tagging action

### PR DESCRIPTION
**Pull request by chase-crumbaugh**
To be used to add tags to codeSamples (or schemas) on merge to main

Relies on new `speakeasy tag promote` command

### CODE REVIEW

**Review by ryan-timothy-albert — APPROVED** [review]
What might also help is just a link to an example repo that implements this action. Specifically what a trigger might look like (presuming we trigger on main)

File: action.yml [hunk 1]
```diff
@@ -19,9 +19,15 @@ inputs:
...
     required: false
+  sources:
+    description: "The sources to tag (comma or newline separated)"
+    required: false
   target:
     description: "Generate a specific target by name"
     required: false
+  code_samples:
+    description: "The targets to tag code samples for (comma or newline separated)"
+    required: false
```
Comment (mfbx9da4) [inline action.yml:30]: I don't understand what this file is for, could you give me some context?

File: internal/cli/speakeasy.go [hunk 2]
```diff
@@ -233,3 +233,30 @@ func CheckFreeUsageAccess() (bool, error) {
...
+
+	if len(sources) > 0 {
+		args = append(args, "-s", strings.Join(sources, ","))
+	}
+	if len(codeSamples) > 0 {
+		args = append(args, "-c", strings.Join(codeSamples, ","))
+	}
+
+	args = append(args, "-t", strings.Join(tags, ","))
+	out, err := runSpeakeasyCommand("tag")
```
Comment (mfbx9da4) [inline internal/cli/speakeasy.go:255]: Did you forget to actually pass through the args here?

Comment (mfbx9da4) [inline internal/cli/speakeasy.go:255]:
```suggestion
	out, err := runSpeakeasyCommand("tag", args)
```

Did you forget to actually pass through the args here?

**Review by mfbx9da4 — APPROVED** [review]

File: .github/workflows/tag.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+name: Speakeasy Registry Tagger
+
+on:
```
Comment (mfbx9da4) [inline .github/workflows/tag.yaml:3]: How will we trigger this on merge to main?

File: internal/cli/speakeasy.go (hunk 2 above)
Comment (chase-crumbaugh) [reply internal/cli/speakeasy.go:255]: Oh lol, 😅

File: .github/workflows/tag.yaml (hunk 3 above)
Comment (chase-crumbaugh) [reply .github/workflows/tag.yaml:3]: Just a standard GH action trigger right?

File: action.yml (hunk 1 above)
Comment (chase-crumbaugh) [reply action.yml:30]: If you had a monorepo for example, you could set up the action to only apply the tag to the codesamples for a specific target

**Pull request merged**